kubemarine/procedures/restore.py

```diff
@@ -135,19 +136,30 @@ def restore_thirdparties(cluster):
 
     install.system_prepare_thirdparties(cluster)
 
-
+    
 def import_nodes(cluster):
-    for node in cluster.nodes['all'].get_ordered_members_list(provide_node_configs=True):
-        node['connection'].put(os.path.join(cluster.context['backup_tmpdir'], 'nodes_data', '%s.tar.gz' % node['name']),
-                               '/tmp/kubemarine-backup.tar.gz')
-        cluster.log.debug('Backup \'%s\' uploaded' % node['name'])
-
-    cluster.log.debug('Unpacking backup...')
-    result = cluster.nodes['all'].sudo(
-        'chattr -i /etc/resolv.conf; sudo tar xzvf /tmp/kubemarine-backup.tar.gz -C / --overwrite && sudo chattr +i /etc/resolv.conf')
-    cluster.log.debug(result)
-
 
+    # Get all nodes
+    nodes = cluster.nodes['all'].get_ordered_members_list(provide_node_configs=True)
+
+    # Upload and restore backups for each node
+    for node in nodes:
+        node_backup_file = os.path.join(cluster.context['backup_tmpdir'], 'nodes_data', '%s.tar.gz' % node['name'])
+        if os.path.exists(node_backup_file):
+            node['connection'].put(node_backup_file, '/tmp/kubemarine-backup.tar.gz')
+            cluster.log.debug('Backup \'%s\' uploaded' % node['name'])
+
+            restore_command = 'sudo tar xzvf /tmp/kubemarine-backup.tar.gz -C / --overwrite'
+            
+            if 'resolv.conf' in tarfile.open(node_backup_file, 'r:gz').getnames():
```

Comment:
The idea is good, but this does not work in case resolv.conf is present in the archive.

Reply:
Okay To check if resolv.conf was extracted separately, i modified the code to search for the file in the tar archive and check if it was extracted to a different path than /etc/resolv.conf. Here's an updated implementation that does this: